mne/time_frequency/tests/test_spectrum.py

```diff
@@ -295,44 +439,135 @@ def _check_spectrum_equivalent(spect1, spect2, tmp_path):
     assert_array_equal(spect1.freqs, spect2.freqs)
 
 
-def test_spectrum_array_errors(epochs_spectrum):
-    """Test EpochsSpectrumArray constructor errors."""
-    data, freqs = epochs_spectrum.get_data(return_freqs=True)
-    info = epochs_spectrum.info
-    with pytest.raises(ValueError, match="Data must be a 3D array"):
-        EpochsSpectrumArray(np.empty((2, 3, 4, 5)), info, freqs)
+@pytest.mark.parametrize("kind", ("raw", "epochs"))
+@pytest.mark.parametrize(
+    "method, output, average",
+    [
+        ("welch", "power", "mean"),  # test with precomputed spectrum
+        ("welch", "power", False),  # unaggregated segments
+        ("multitaper", "complex", None),  # unaggregated tapers
+    ],
+)
+def test_spectrum_array_errors(kind, method, output, average, request):
+    """Test (Epochs)SpectrumArray constructor errors."""
+    if method == "welch" and output == "power" and average:
+        spectrum = request.getfixturevalue(f"{kind}_spectrum")
+    else:
+        data = request.getfixturevalue(kind)
+        kwargs = dict()
+        if method == "welch":
+            kwargs.update(average=average)
+        spectrum = data.compute_psd(method=method, output=output, **kwargs)
+    data, freqs = spectrum.get_data(return_freqs=True)
+    info = spectrum.info
+    mt_weights = spectrum.mt_weights
+    Klass = SpectrumArray if kind == "raw" else EpochsSpectrumArray
+    # test mismatching number of channels
+    bad_n_chans = data[:-1] if kind == "raw" else data[:, :-1]
     with pytest.raises(ValueError, match=r"number of channels.*good data channels"):
-        EpochsSpectrumArray(data[:, :-1], info, freqs)
-    with pytest.raises(ValueError, match=r"last dimension.*same number of elements"):
-        EpochsSpectrumArray(data[..., :-1], info, freqs)
+        Klass(bad_n_chans, info, freqs)
+    # test mismatching number of frequencies
+    bad_n_freqs = (
+        data[..., :-1, :] if method == "welch" and not average else data[..., :-1]
+    )
+    with pytest.raises(ValueError, match=r"number of frequencies.*number of elements"):
+        Klass(bad_n_freqs, info, freqs, method=method, mt_weights=mt_weights)
     # test mismatching events shape
-    n_epo = data.shape[0] + 1  # +1 so they purposely don't match
-    events = np.vstack(
-        (np.arange(n_epo), np.zeros(n_epo, dtype=int), np.ones(n_epo, dtype=int))
-    ).T
-    with pytest.raises(ValueError, match=r"first dimension.*dimension of `events`"):
-        EpochsSpectrumArray(data, info, freqs, events)
+    if kind == "epochs":
```

Comment:
not testing with Raw also avoids this `if kind` logic, which simplifies the (already quite long) test slightly.

Reply:
Yeah, have returned to Epochs only